Full text search

tylerwince opened the pull request:
## Description

This PR adds the ability for a user to search tags, search OCR Text, search OCR attachments, and search full note body text in addition to the note title. 


## Screencast

<img width="918" alt="CleanShot 2023-05-18 at 17 30 42@2x" src="https://github.com/raycast/extensions/assets/17678033/04e73e2c-9037-423b-b2ff-04bce9ea5993">


## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

tylerwince commented:
@mathieudutour any ideas how I get the typescript definitions check to pass? The code that is failing is generated code from the protocol buffer definition so it can’t really be changed.

StevenRCE0 commented:
https://github.com/raycast/extensions/blob/8df6a07ce34e7a6b745b492e72bff5ba2285bc8e/extensions/apple-notes/src/proto/notestore.ts#L2205-L2206

Have you tried this?

mathieudutour commented:
> The code that is failing is generated code from the protocol buffer definition so it can’t really be changed

How is it generated and why couldn't you change it?

tylerwince commented:
> > The code that is failing is generated code from the protocol buffer definition so it can’t really be changed
> 
> How is it generated and why couldn't you change it?

It is generated via protoc. Essentially, Apple Notes leverages Protocol Buffers to serialize and store the note content (both text and images/attachments/etc). So, when serializing and deserializing protocol buffers, you define the data format in a `.proto` file (you can see that one [here](https://github.com/raycast/extensions/blob/8df6a07ce34e7a6b745b492e72bff5ba2285bc8e/extensions/apple-notes/src/proto/notestore.proto). 

The proto file defines the data, and then you generate the code to serialize and deserialize using a [compiler](https://protobuf.dev/programming-guides/proto2/#generating). It probably doesn't makes sense to rewrite this code by hand since it's auto-created from that `.proto` file.

mathieudutour commented:
While you don't have to rewrite the code generator, you can just manually add a ts-ignore, no?

tylerwince commented:
For sure - I'm good to do that if that's the route Raycast would like. I just didn't want to add a ts-ignore to generated code if it wasn't desired by the team. I added that and updated the PR!

Let me know if you have any other feedback, happy to support any changes

pernielsentikaer commented:
@tylerwince do you mind adding a changelog entry

StevenRCE0 commented:
> Do you have anything else before merging this @StevenRCE0

Nah, it looks very promising!

pernielsentikaer commented:
@tylerwince it also looks like this will change some other behaviors when opening notes, is that intended?
